dev/tools/test/content_aware_hash_test.dart

```diff
@@ -0,0 +1,335 @@
+        print('$prefix:>$s<');
+      });
+    }
+  }
+
+  io.ProcessResult run(String executable, List<String> args, {String? workingPath}) {
+    print('Running "$executable ${args.join(" ")}"${workingPath != null ? ' $workingPath' : ''}');
+    final io.ProcessResult result = io.Process.runSync(
+      executable,
+      args,
+      environment: environment,
+      workingDirectory: workingPath ?? testRoot.root.absolute.path,
+      includeParentEnvironment: false,
+    );
+    if (result.exitCode != 0) {
+      fail(
+        'Failed running "$executable $args" (exit code = ${result.exitCode}),'
+        '\nstdout: ${result.stdout}'
+        '\nstderr: ${result.stderr}',
+      );
+    }
+    printIfNotEmpty('stdout', (result.stdout as String).trim());
+    printIfNotEmpty('stderr', (result.stderr as String).trim());
+    return result;
+  }
+
+  setUp(() async {
+    tmpDir = localFs.systemTempDirectory.createTempSync('content_aware_hash.');
+    testRoot = _FlutterRootUnderTest.fromPath(tmpDir.childDirectory('flutter').path);
+
+    environment = <String, String>{};
+
+    if (const LocalPlatform().isWindows) {
+      // Copy a minimal set of environment variables needed to run the update_engine_version script in PowerShell.
+      const List<String> powerShellVariables = <String>['SystemRoot', 'Path', 'PATHEXT'];
+      for (final String key in powerShellVariables) {
+        final String? value = io.Platform.environment[key];
+        if (value != null) {
+          environment[key] = value;
+        }
+      }
+    }
+
+    // Make a slim copy of the flutterRoo
+    flutterRoot.copyTo(testRoot);
+  });
+
+  tearDown(() {
+    // Git adds a lot of files, we don't want to test for them.
+    final Directory gitDir = testRoot.root.childDirectory('.git');
+    if (gitDir.existsSync()) {
+      gitDir.deleteSync(recursive: true);
+    }
+
+    // Now do cleanup so even if the next step fails, we still deleted tmp.
+    tmpDir.deleteSync(recursive: true);
+  });
+
+  /// Runs `bin/internal/content_aware_hash.{sh|ps1}` and returns the process result.
+  ///
+  /// If the exit code is 0, it is considered a success, and files should exist as a side-effect.
+  ///   - On Windows, `powershell` is used (to run `update_engine_version.ps1`);
+  ///   - Otherwise, `update_engine_version.sh` is used.
+  io.ProcessResult runContentAwareHash() {
+    final String executable;
+    final List<String> args;
+    if (const LocalPlatform().isWindows) {
+      executable = 'powershell';
+      args = <String>[testRoot.contentAwareHashPs1.path];
+    } else {
+      executable = testRoot.contentAwareHashSh.path;
+      args = <String>[];
+    }
+    return run(executable, args);
+  }
+
+  /// Initializes a blank git repo in [testRoot.root].
+  void initGitRepoWithBlankInitialCommit({String? workingPath}) {
+    run('git', <String>['init', '--initial-branch', 'master'], workingPath: workingPath);
+    run('git', 'config --local core.autocrlf true'.split(' '));
```

Comment:
codefusay: missing comment - did it not get pushed? autocrlf is important for testing.

Reply:
codefu also say: add workingPath: workingPath this.